compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs

```diff
@@ -146,13 +146,13 @@ namespace std
         #{initMetadata}
     };
 
-    #{template}inline void swap(#{structName}& left, #{structName}& right)
+    #{template}inline void swap(#{qualifiedStructName}& left, #{qualifiedStructName}& right)
     {
         left.swap(right);
     }|]
       where
         template = CPP.template s
-        structName = CPP.structName s
+        qualifiedStructName = CPP.qualifiedStructName cpp s
```

Comment:
Done. I looked at doing it in `TypeMapping`, and I sorta see how it will work, but given that we don't have a second use for this yet, I don't think it's worth the time to figure it out right now.